packages/core/src/renderTexture/RenderTextureSystem.ts

```diff
@@ -155,8 +155,8 @@ export class RenderTextureSystem extends System
 
             if (!sourceFrame)
             {
-                tempRect.width = renderer.screen.width;
-                tempRect.height = renderer.screen.height;
+                tempRect.width = renderer.view.width / renderer.resolution;
```

Comment:
This seems unnecessary - `screen.width` should always equal `view.width / renderer.resolution`.

Reply:
Now always, because the framebuffer dimensions can't be noninteger of course. For example `width = 101` and `resolution = 1.5`: the framebuffer can't be `151.5`, so it becomes `151` instead (https://www.pixiplayground.com/#/edit/ovoCtZn-zaHjGp1ki5bAJ). That's why `screen.width` can be slightly larger in some cases. Small difference, but, if we omit this change, we get a slightly blurry image.

Reply:
`renderer.view.width` should be `151.5` in your case. It isn't rounded.

https://github.com/pixijs/pixi.js/blob/3f048344f63a436bad40b0d0542a34bd8dc692b9/packages/core/src/AbstractRenderer.ts#L273

Reply:
Non-integer resolutions are not supported for this reason anyway. Maybe we should throw an error if the `view.width` turns out to be a non-integer and make that explicit.

Reply:
> 
> 
>

> Non-integer resolutions are not supported for this reason anyway. Maybe we should throw an error if the `view.width` turns out to be a non-integer and make that explicit.

Isn't `window.devicePixelRatio` used as resolution for the renderer in many cases? That would break a lot of code.

> 
> 
> `renderer.view.width` should be `151.5` in your case. It isn't rounded.
> 
> https://github.com/pixijs/pixi.js/blob/3f048344f63a436bad40b0d0542a34bd8dc692b9/packages/core/src/AbstractRenderer.ts#L273

Check the assert's in the playground example. It's `151`. It isn't rounded by pixi itself but in `HTMLCanvasElement` or elsewhere.

Reply:
Okay, so (/cc @ivanpopelyshev) I think we should strictly enforce resolutions that cause non-integer framebuffer dimensions.

Reply:
> This seems unnecessary - screen.width should always equal view.width / renderer.resolution.

We should support this thing. That's why i mark all those issues as X-Files, it takes a bit of fiddling to get floors/ceils/rounds right

Reply:
We could/should change `AbstractRenderer.resize()`:

```js
    /**
     * Resizes the screen and canvas as close as possible to the specified width and height.
     * Canvas dimensions are multiplied by resolution and rounded to the nearest integers.
     * The new canvas dimensions divided by the resolution become the new screen dimensions.
     *
     * @param desiredScreenWidth - The desired width of the screen.
     * @param desiredScreenHeight - The desired height of the screen.
     */
    resize(desiredScreenWidth: number, desiredScreenHeight: number): void
    {
        this.view.width = Math.round(desiredScreenWidth * this.resolution);
        this.view.height = Math.round(desiredScreenHeight * this.resolution);

        const screenWidth = this.view.width / this.resolution;
        const screenHeight = this.view.height / this.resolution;

        this.screen.width = screenWidth;
        this.screen.height = screenHeight;

        if (this.autoDensity)
        {
            this.view.style.width = `${screenWidth}px`;
            this.view.style.height = `${screenHeight}px`;
        }

        /**
         * Fired after view has been resized.
         *
         * @event PIXI.Renderer#resize
         * @param {number} screenWidth - The new width of the screen.
         * @param {number} screenHeight - The new height of the screen.
         */
        this.emit('resize', screenWidth, screenHeight);
    }
```